src/v/cluster/tm_stm.h

```diff
@@ -66,6 +61,42 @@ class tm_stm final : public persisted_stm {
         timeout
     };
 
+    struct tm_tx_hash_ranges {
+        using hash_ranges_type = std::vector<std::pair<uint32_t, uint32_t>>;
```

Comment:
Please introduce an explicit struct instead of pair and add comments there on what the fields mean e.g. if the interval is open or closed

Reply:
done